File: server/internal/controller/system/captcha.go
// +----------------------------------------------------------------------
// | XYGo Admin [ Vue3 + GoFrame 企业级中后台管理系统 ]
// +----------------------------------------------------------------------
// | Copyright (c) 2026 大连星韵网络科技有限公司 All rights reserved.
// +----------------------------------------------------------------------
// | Licensed ( https://opensource.org/licenses/MIT )
// +----------------------------------------------------------------------
// | Author: 喜羊羊 <751300685@qq.com>
// +----------------------------------------------------------------------

package system

import (
	"context"

	"github.com/gogf/gf/v2/errors/gerror"

	api "xygo/api/system"
	"xygo/internal/library/captcha"
)

// ClickCaptcha 获取点选验证码
func (c *ControllerV1) ClickCaptcha(ctx context.Context, req *api.ClickCaptchaReq) (res *api.ClickCaptchaRes, err error) {
	result, err := captcha.GenerateClick(ctx)
	if err != nil {
		return nil, err
	}
	return &api.ClickCaptchaRes{
		Id:     result.Id,
		Text:   result.Text,
		Base64: result.Base64,
		Width:  result.Width,
		Height: result.Height,
	}, nil
}

// CheckClickCaptcha 校验点选验证码
func (c *ControllerV1) CheckClickCaptcha(ctx context.Context, req *api.CheckClickCaptchaReq) (res *api.CheckClickCaptchaRes, err error) {
	if !captcha.VerifyClick(ctx, req.Id, req.Info) {
		return nil, gerror.New("验证失败，请重试")
	}
	return &api.CheckClickCaptchaRes{}, nil
}

File: web/src/api/common/captcha.ts
// +----------------------------------------------------------------------
// | XYGo Admin [ Vue3 + GoFrame 企业级中后台管理系统 ]
// +----------------------------------------------------------------------
// | Copyright (c) 2026 大连星韵网络科技有限公司 All rights reserved.
// +----------------------------------------------------------------------
// | Licensed ( https://opensource.org/licenses/MIT )
// +----------------------------------------------------------------------
// | Author: 喜羊羊 <751300685@qq.com>
// +----------------------------------------------------------------------

/**
 * 点选验证码公共 API（前后台通用）
 * 接口无需鉴权，路径: /captcha/click, /captcha/checkClick
 */
import api from '@/utils/http'

/** 点选验证码响应 */
export interface ClickCaptchaResult {
  id: string
  text: string[]
  base64: string
  width: number
  height: number
}

/**
 * 获取点选验证码
 */
export function getClickCaptcha() {
  return api.get<ClickCaptchaResult>({
    url: '/captcha/click'
  })
}

/**
 * 校验点选验证码
 */
export function checkClickCaptcha(id: string, info: string) {
  return api.post<void>({
    url: '/captcha/checkClick',
    data: { id, info }
  })
}
